CHANGELOG.md

```diff
@@ -1,5 +1,173 @@
+```python
+pytest_plugins = 'aiida.tools.pytest_fixtures'
+```
+The new fixtures include the `aiida_profile` fixture which is session-scoped and automatically loaded.
+The fixture creates a temporary test profile at the start of the test session and automatically deletes it when the session ends.
+For more information and an overview of all available fixtures, please refer to [the documentation on `pytest` fixtures](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/plugins.html#plugin-test-fixtures).
+
+### Improvements and changes to caching
+
+A number of fixes and changes to the caching mechanism were introduced (see the [changes](#changes) subsection of the [full list of changes](#full-list-of-changes) for a more detailed overview).
+As a result, a database migration had to be added in order to reset all existing hashes.
+You will be automatically prompted to migrate your database after updating your installation.
+After the migration has been completed, you can use `verdi node rehash` to recreate the hashes according to the new algorithm and repopulate the cache.
+
+Since its introdcuction, the cache would essentially be reset each time AiiDA or any of the plugin packages would be updated, since the version of these packages were included in the calculation of the node hashes.
+This was originally done out of precaution to err on the safe-side and limit the possibility of false-positives in cache hits.
+However, this strategy has turned out to be unnecessarily cautious and severely limited the effectiveness of caching.
+
+The package version information is no longer included in the hash and therefore no longer impacts the caching.
+This change does now make it possible for false positives if the implementation of a `CalcJob` or `Parser` plugin changes signficantly.
+Therefore, a mechanism is introduced to give control to these plugins to effectively reset the cache of existing nodes.
+Please refer to the [documentation on controlling caching](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/provenance/caching.html#calculation-jobs-and-parsers) for more details.
+
+### Programmatic syntax for query builder filters and projections
+
+In the `QueryBuilder`, fields to filter on or project always had to be provided with strings:
+```python
+QueryBuilder().append(Node, filters={'label': 'some-label'}, project=['id', 'ctime'])
+```
+and it is not always trivial to know what fields exist that _can_ be filtered on and can be projected.
+In addition, there was a discrepancy for some fields, most notably the `pk` property, which had to be converted to `id` in the query builder syntax.
+
+These limitations have been solved as each class in AiiDA's ORM now defines the `fields` property, which allows to discover these fields programmatically.
+The example above would convert to:
+```python
+QueryBuilder().append(Node, filters={'label': Node.fields.label}, project=[Node.fields.pk, Node.fields.ctime])
```

Comment:
Is this example correct? I don't know much about this feature but I would expect it should be

```suggestion
QueryBuilder().append(Node, filters={Node.fields.label: 'some-label'}, project=[Node.fields.pk, Node.fields.ctime])
```